src/v/storage/log.h

```diff
@@ -181,13 +182,24 @@ class log final {
         return _impl->stm_manager();
     }
     /**
-     * Controlls the max offset that may be evicted by log retention policy
-     * This offset is non-decreasing.
+     * Controlls the max offset that may be compacted or evicted by log
+     * retention policy. This offset is non-decreasing.
      */
     void set_collectible_offset(model::offset o) {
         return _impl->set_collectible_offset(o);
     };
 
+    /**
+     * We currently use `max_collectible_offset` to control both
+     * deletion/eviction, and compaction.
+     */
+    static bool
+    has_compactible_offsets(const segment& seg, const compaction_config& cfg) {
+        // since we don't support partially-compacted segments, a segment must
+        // end before the max compactible offset to be eligible for compaction.
+        return seg.offsets().stable_offset <= cfg.max_collectible_offset;
+    }
+
```

Comment:
Maybe extract to util function this is not really related to log, we should not use segments in the `log` abstraction as segments are valid only for disk_log_impl

Reply:
maybe make it a `segment` method ?

Reply:
Ah.. I see. I thought segment was generic as well. I like moving it to `segment`, thanks.

Reply:
Done in latest update.